source/extensions/geoip_providers/maxmind/geoip_provider.cc

```diff
@@ -150,6 +214,12 @@ void GeoipProvider::lookupInCityDb(
           populateGeoLookupResult(mmdb_lookup_result, lookup_result,
                                   config_->countryHeader().value(), MMDB_COUNTRY_LOOKUP_ARGS[0],
                                   MMDB_COUNTRY_LOOKUP_ARGS[1]);
+          auto it = lookup_result.find(config_->countryHeader().value());
+          std::string country;
+          if (it != lookup_result.end()) {
+            country = it->second;
+          }
+          lookup_result[config_->countryHeader().value()] = getCountryMapping(country);
```

Comment:
highly likely this assignment operator will copy the string value of `getCountryMapping(country)` call, you could use move semantics (capture getCountryMapping(country) by rvalue) instead to avoid extra copy: https://cplusplus.com/reference/string/string/operator=/